requirements-dev.txt

```diff
@@ -1,7 +1,7 @@
 coverage>=4.4.0
 matplotlib>=2.1
 black[jupyter]~=22.0
-pylint>=2.8.3,<2.14.0
+pylint>=2.15.0
```

Comment:
In the apps repos we went to `mypy>=0.981` when we added 3.11. Why that version I am not sure. Since we always install the latest in CI arguably it does not matter too much.

The other thing I note, seeing sphinx versions and autodoc-typehints here is that at some point we could update the docs build here to use the built-in typehint support as was done as part of this qiskit-community/qiskit-nature#1109

Reply:
Yeh I was hoping that once the namespace migration has settled in the other repos and once we can assume that the docs have stabilized to a satisfying degree we can make sure to synchronize this with here, too :+1: 

The same goes for such tooling versions. I am not sure why the upper bound got raised there :thinking: